app/src/main/java/com/infomaniak/swisstransfer/ui/OnboardingActivity.kt

```diff
@@ -17,13 +17,51 @@
  */
 package com.infomaniak.swisstransfer.ui
 
+import android.content.Intent
+import android.graphics.Color
+import android.os.Build
 import android.os.Bundle
 import androidx.activity.ComponentActivity
+import androidx.activity.SystemBarStyle
+import androidx.activity.compose.setContent
 import androidx.activity.enableEdgeToEdge
+import androidx.compose.material3.Surface
+import androidx.compose.runtime.rememberCoroutineScope
+import com.infomaniak.swisstransfer.ui.screen.onboarding.OnboardingScreen
+import com.infomaniak.swisstransfer.ui.theme.SwissTransferTheme
+import com.infomaniak.swisstransfer.ui.utils.AccountUtils
+import dagger.hilt.android.AndroidEntryPoint
+import kotlinx.coroutines.launch
+import javax.inject.Inject
 
+@AndroidEntryPoint
 class OnboardingActivity : ComponentActivity() {
+
+    @Inject
+    lateinit var accountUtils: AccountUtils
+
     override fun onCreate(savedInstanceState: Bundle?) {
-        enableEdgeToEdge()
         super.onCreate(savedInstanceState)
+        enableEdgeToEdge(navigationBarStyle = SystemBarStyle.auto(Color.TRANSPARENT, Color.TRANSPARENT))
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) window.isNavigationBarContrastEnforced = false
+
+        setContent {
+            val scope = rememberCoroutineScope()
+
+            SwissTransferTheme {
+                Surface {
+                    OnboardingScreen(
+                        goToMainActivity = {
+                            scope.launch {
+                                accountUtils.login()
+                                startActivity(Intent(this@OnboardingActivity, MainActivity::class.java))
```

Comment:
Could we rather simplify the writing of trivial `startActivity`s with something like this instead if we want an easier way of reading code?:

```kt
startActivity(MainActivity::class)
```

This can be achieved using the following method:

```kt
fun <T : Activity> Context.startActivity(targetActivity: KClass<T>) {
    startActivity(Intent(this, targetActivity.java))
}
```